codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/AcceptorGenerator.kt

```diff
@@ -83,18 +83,18 @@ private fun KotlinWriter.renderPathAcceptor(wi: WaiterInfo, directive: String, i
     withBlock("#T(RetryDirective.#L) {", "},", acceptorType, directive) {
         val visitor = KotlinJmespathExpressionVisitor(wi.ctx, this, visitedShape)
         val expression = JmespathExpression.parse(matcher.path)
-        val (actual, _) = expression.accept(visitor)
+        val actual = expression.accept(visitor)
 
         val expected = matcher.expected
         val comparison = when (matcher.comparator!!) {
-            PathComparator.STRING_EQUALS -> "$actual == ${expected.dq()}"
-            PathComparator.BOOLEAN_EQUALS -> "$actual == ${expected.toBoolean()}"
-            PathComparator.ANY_STRING_EQUALS -> "$actual?.any { it == ${expected.dq()} } ?: false"
+            PathComparator.STRING_EQUALS -> "${actual.identifier} == ${expected.dq()}"
+            PathComparator.BOOLEAN_EQUALS -> "${actual.identifier} == ${expected.toBoolean()}"
+            PathComparator.ANY_STRING_EQUALS -> "${actual.identifier}?.any { it == ${expected.dq()} } ?: false"
```

Comment:
**Comment**: Friendly reminder that data classes _do_ support positional destructuring so you can keep the tuple assignment if that makes the code cleaner:

```kotlin
val (actual, _) = expression.accept(visitor)

...

PathComparator.STRING_EQUALS -> "$actual == ${expected.dq()}"
PathComparator.BOOLEAN_EQUALS -> "$actual == ${expected.toBoolean()}"
PathComparator.ANY_STRING_EQUALS -> "$actual?.any { it == ${expected.dq()} } ?: false"